pywr/recorders/_recorders.pyx

```diff
@@ -1042,6 +1054,19 @@ cdef class FlowDurationCurveDeviationRecorder(FlowDurationCurveRecorder):
         else:
             return df
 
+    @classmethod
+    def load(cls, model, data):
+        node = model.nodes[data.pop("node")]
+        percentiles = data.pop("percentiles")
+        from pywr.parameters import load_parameter_values
+        upper_target_fdc = data.pop("upper_target_fdc", None)
```

Comment:
Should these use the `percentiles` to get the values from the dataframe? E.g. if you specified the percentiles in a different order, or had more values defined in the dataframe?

Reply:
Something like,

```python
upper_target_fdc = upper_target_fdc.loc[percentiles, :]
```

Reply:
Done but as `load_parameter_values` returns a numpy array I had to make a small update to the dataframe functions

Reply:
OK, are you happy with this now basically requiring the input data to be indexed?

Reply:
yes, I think its a good check and not to onerous for anyone to do. I added comments to the docstring stating that this is necessary.